index.js

```diff
@@ -24,64 +26,93 @@ function getRandomNumber(min, max) {
 
 function checkGuess() {
   // Get value from guess input element
-  const guess = parseInt(guessInput.value, 10);
-  attempts = attempts + 1;
-
+  let guess = parseInt(guessInput.value, 10);
+  guessInput.value='';
+  
   hideAllMessages();
-
+  resetButton.style.display = ''
+  
   if (guess === targetNumber) {
-    numberOfGuessesMessage.style.display = '';
     numberOfGuessesMessage.innerHTML = `You made ${attempts} guesses`;
-
+    numberOfGuessesMessage.style.display='block';
     correctMessage.style.display = '';
-
     submitButton.disabled = true;
+    
     guessInput.disabled = true;
+    
+    resetButton.style.display = '';
+    
   }
+  
+   
 
-  if (guess !== targetNumber) {
-    if (guess < targetNumber) {
-      tooLowMessage.style.display = '';
-    } else {
-      tooLowMessage.style.display = '';
-    }
+  if (guess < targetNumber) {
+    tooLowMessage.style.display = '';
+    
+  }
+  
+   
+   
+  if (guess > targetNumber) {
+    tooHighMessage.style.display = '';
+    
+  }
 
-    const remainingAttempts = maxNumberOfAttempts - attempts;
 
-    numberOfGuessesMessage.style.display = '';
-    numberOfGuessesMessage.innerHTML = `You guessed ${guess}. <br> ${remainingAttempts} guesses remaining`;
-  }
+  const remainingAttempts = maxNumberOfAttempts - attempts;
+  console.log(remainingAttempts);
 
-  if (attempts ==== maxNumberOfAttempts) {
-    submitButton.disabled = true;
-    guessInput.disabled = true;
+  if(guess !==targetNumber){
+    
+
+    maxGuessesMessage.innerHTML = `You guessed ${guess} <br>   ${remainingAttempts} guesses remaining`;
+  maxGuessesMessage.style.display = '';
+  
+    
+}
+
+else{
+  document.getElementById('max-guesses');
+  
+  //submitButton.disabled=true;
+   //guessInput.disabled=true;
+ }
+ if(attempts===maxNumberOfAttempts){
+  guessInput.disabled=true;
+  submitButton.disabled=true;
   }
+    attempts++;
```

Comment:
Right now, if I put in incorrect guesses, it actually lets me put in 6 guesses instead of just 5. That's because you don't increment the `attempts` variable until _after_ you check if you need to show the "remainingAttempts" text or disabling the guess button and the submit button.

For example, on the first guess:
- `attempts` would be equal to 0 since that's the value it starts with
- the calculation on line 64 (`const remainingAttempts = maxNumberOfAttempts - attempts;`) would set `remainingAttempts` to be `5 - 0`, so `5`. So the text says "You have 5 guesses remaining". Which isn't actually true; I just made a guess so there should only be four guesses remaining.

Same with if I guess it correctly on my first try. On line 36, it will say "You made 0 guesses". It _should_ say, "You made 1 guesses". 

You can fix this by moving the `attempts++` to earlier on in the function before things try to use that variable; for example onto line `34` before the `if (guess === targetNumber)` block.

Reply:
so after 1 guess it should say 4 guesses remaining?